view/frontend/web/js/view/payment/method-renderer/bold-wallet-payments.js

```diff
@@ -0,0 +1,72 @@
+define(
+    [
+        'Magento_Checkout/js/view/payment/default',
+        'ko',
+        'Bold_CheckoutPaymentBooster/js/model/spi',
+    ], function (
+        DefaultPaymentComponent,
+        ko,
+        spi,
+    ) {
+        'use strict';
+        return DefaultPaymentComponent.extend({
+            defaults: {
+                template: 'Bold_CheckoutPaymentBooster/payment/wallet-payments',
+                isVisible: ko.observable(false),
+                isWalletPayLoading: ko.observable(false),
+            },
+            /** @inheritdoc */
+            initialize: function () {
+                this._super(); //call Magento_Checkout/js/view/payment/default::initialize()
+                this.isVisible.subscribe((isVisible) => {
+                    if (isVisible) {
+                        this.initPaymentButtons();
+                    }
+                });
+                this.isVisible(window.checkoutConfig.bold?.paymentBooster);
+            },
+            /**
+             * Initialize Payment Buttons.
+             *
+             * @return {Promise<void>}
+             */
+            initPaymentButtons: async function () {
+                const observer = new MutationObserver(function () {
+                    if (document.getElementById('wallet-payments')) {
+                        observer.disconnect();
+                        this.renderPaymentsButtons();
+                    }
+                }.bind(this));
+                observer.observe(document.body, {
+                    childList: true,
+                    subtree: true,
+                });
+            },
+            /**
+             * Load Payment Buttons.
+             *
+             * @returns {Promise<void>}
+             */
+            renderPaymentsButtons: async function () {
+                this.isWalletPayLoading(true);
+                const boldPaymentsInstance = await spi.getPaymentsClient();
+                const gatewayData = boldPaymentsInstance.paymentGateways[0] || null;
+                const isPaymentsButtonsVisible = gatewayData && (gatewayData.type === 'ppcp' || (gatewayData.type === 'braintree' && (gatewayData.credentials.is_paypal_enabled || gatewayData.credentials.is_google_pay_enabled || gatewayData.credentials.is_paypal_enabled)));
```

Comment:
This is a lengthy assertion, could we format this on multiple lines for readability and future maintainability

Reply:
fixed